addon/components/file-browser-icon/component.js

```diff
@@ -0,0 +1,31 @@
+import Ember from 'ember';
+import layout from './template';
+
+export default Ember.Component.extend({
+  layout,
+  tagName: 'span',
+  classNames: ['file-browser-icon'],
+
+  click() {
+      if (this.get('row.isExpandable')) {
+          this.toggleProperty('row.expanded');
+      }
+  },
+
+  iconName: Ember.computed('row', 'row.expanded', function() {
```

Comment:
Did you consider using [`classNameBindings`](http://emberjs.com/api/classes/Ember.Component.html#property_classNameBindings) here? Might be more concise?